core/internal/cluster/kafka_cluster.go

```diff
@@ -278,3 +299,38 @@ func (module *KafkaCluster) getOffsets(client helpers.SaramaClient) {
 		return false
 	})
 }
+
+func (module *KafkaCluster) reapNonExistingGroups(client helpers.SaramaClient) {
+	kafkaGroups, err := client.ListConsumerGroups()
+	if err != nil {
+		module.Log.Warn("failed to get the list of available consumer groups", zap.Error(err))
```

Comment:
I would expect a return here to stop the method.

Reply:
Also, why Warn over Error?

Reply:
I added the return and changed it to use Error